Sources/MockoloFramework/Parsers/SwiftSyntaxExtensions.swift

```diff
@@ -738,42 +751,77 @@ final class EntityVisitor: SyntaxVisitor {
+                }
+                previousConditions.append(conditionDescription)
             } else {
-                return .visitChildren
+                if let previousCondition = previousConditions.first {
+                    compilerDirectiveKey = "\(previousCondition):\(blockId):else"
+                } else {
+                    return .visitChildren
+                }
             }
 
-            guard macroName != fileMacro else { return .visitChildren }
-
             if let list = cl.elements?.as(CodeBlockItemListSyntax.self) {
                 for el in list {
+                    let nestedImport: ImportStatement
                     if let importItem = el.item.as(ImportDeclSyntax.self) {
-                        let key = macroName
-                        if imports[key] == nil {
-                            imports[key] = []
-                        }
-                        imports[key]?.append(importItem.trimmedDescription)
-
+                        nestedImport = .init(
+                            line: importItem.trimmedDescription,
+                            compilerDirectiveKey: compilerDirectiveKey
+                        )
                     } else if let nested = el.item.as(IfConfigDeclSyntax.self) {
-                        let key = macroName
-                        if imports[key] == nil {
-                            imports[key] = []
+                        var nestedImportLines: String = ""
+                        for (index, clause) in nested.clauses.enumerated() {
+                            let clausePrefix: String
+                            if let nestedCondition = clause.condition?.trimmedDescription {
+                                clausePrefix = index == 0 ? "#if \(nestedCondition)" : "#elseif \(nestedCondition)"
+                            } else {
+                                clausePrefix = "#else"
+                            }
+                            if let items = clause.elements?.as(CodeBlockItemListSyntax.self) {
+                                let importLines = items
+                                    .compactMap({ $0.item.as(ImportDeclSyntax.self) })
+                                    .map(\.trimmedDescription)
+                                    .joined(
+                                        separator: "\n"
+                                    )
+                                nestedImportLines += [
+                                    clausePrefix,
+                                    importLines,
+                                ].joined(separator: "\n")
+                            }
                         }
-                        imports[key]?.append(nested.trimmedDescription)
+                        nestedImportLines.append("\n#endif")
+                        nestedImport = .init(
+                            line: nestedImportLines,
+                            compilerDirectiveKey: compilerDirectiveKey
+                        )
```

Comment:
Here, I aggregate all imports inside the compiler directive and wrap them by clause (such as poundIf, poundElseIf and poundElse).

I think this approach is not best, however I believe model architecture refinement might be required to fix underlying issue.

Fortunately, import statement cannot be written in except for top-level. So I think this approach will not produce bugs. We don't have to visiting further child nodes to find import decl syntaxes.

This implementation does not support nested-nested-imports, though.

Reply:
Oh, it was clearly written here that nesting is not supported.